Review

williamgoulois opened the pull request.

baptistesx committed:
migrate from react to nextjs

baptistesx committed:
fix route protection

baptistesx committed:
fix refresh redirection issue

baptistesx committed:
rename route to pascal-case

baptistesx committed:
catch Error in workaway bot

baptistesx committed:
add selected prop on drawer item

baptistesx committed:
hide toolbar items on responsive

baptistesx committed:
Merge pull request #1 from baptistesx/develop

Develop

baptistesx committed:
remove '/' on api routes

baptistesx committed:
remove '/' on route

baptistesx committed:
some pages refacto

baptistesx committed:
fix all nav issue by adding ssr check

baptistesx committed:
fix + refacto

baptistesx committed:
test

baptistesx committed:
test

baptistesx committed:
fix destination on Home page

baptistesx committed:
Add responsiveFontSizes

baptistesx committed:
test with mock user

baptistesx committed:
test

Commit:
replace *any* param type by the expected types + create a Guard to differenciate google responses

baptistesx committed:
test

baptistesx committed:
add custom 404 error page fix

baptistesx committed:
add app logo + update favicon

baptistesx committed:
update readme

baptistesx committed:
fix navigating to the same current page

baptistesx committed:
wrap button with next Link component

baptistesx committed:
remove SSR on pages

baptistesx committed:
format file

baptistesx committed:
Merge pull request #2 from baptistesx/chore/fix-googleTokenAuth-type

Chore/fix google token auth type

baptistesx committed:
Merge pull request #3 from baptistesx/remove_ssr

remove SSR on pages

baptistesx committed:
fix : remove GlobalLayout user param

baptistesx committed:
Implement AuthProvider + reorganize all imports

baptistesx committed:
fix User import + add refetch user when update with dialog on users page

baptistesx committed:
implement isDirty on forms

baptistesx committed:
remove snackbar error fetching user on NotSignedInRoutes

baptistesx committed:
Display a custom Welcom message if new user or not

baptistesx committed:
refresh user when updated

baptistesx committed:
create SignedInLayout and NotSignedInLayout

baptistesx committed:
Merge pull request #4 from baptistesx/redesign-not-signedin-pages

create SignedInLayout and NotSignedInLayout

baptistesx committed:
Merge pull request #5 from baptistesx/develop

Develop

baptistesx committed:
add sign in button on sign up form

baptistesx committed:
responsive NotSignedInLayout

baptistesx committed:
fix responsive issues

baptistesx committed:
fix logo name in SignedInLayout

baptistesx committed:
refacto roles

baptistesx committed:
Merge pull request #6 from baptistesx/refacto-roles

refacto roles

baptistesx committed:
refacto GetLicenceButton + redesign email validation alert

baptistesx committed:
update error handling for user creation

baptistesx committed:
add update password feature on profile page

baptistesx committed:
add send verification email again feature on dashboard Alert

baptistesx committed:
fix setIsLoading(false) in catch update password

baptistesx committed:
update UI

baptistesx committed:
update paypal for production

baptistesx committed:
add useSWR to refresh the user frequently

williamgoulois commented on .babelrc:
```diff
@@ -0,0 +1,4 @@
+{
```
Tu peux normalement utiliser SWC et supprimer ton babelrc pour de meilleurs perf
https://nextjs.org/docs/advanced-features/compiler#unsupported-features

williamgoulois commented on .env.dist:
```diff
@@ -0,0 +1,5 @@
+REACT_APP_ENDPOINT=
```
À update pour next non ? https://nextjs.org/docs/basic-features/environment-variables
Tu peux créer un fichier process.env.d.ts pour typer tes variables d'env avec TS https://stackoverflow.com/a/53981706/17912798

baptistesx replied:
Pour le fichier de typage des variables d'env, tu proposes "process.env.d.ts" et sur SO "environment.d.ts", du coup, j'utilise quel nom ? x)

En l'occurrence j'utilise l'extension ".dist" qui contient les noms de variables mais sans leurs valeurs (fichier qui est push sur le repo). C'est bien? Si oui, il faut que je check s'il y a un moyen de les garder sync avec les .env.local, .env.development.local et .env.production.local

williamgoulois commented on .eslintrc.json:
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": "next/core-web-vitals"
+  "extends": ["next/babel","next/core-web-vitals"]
```
C'est un bon début mais il faudra à l'avenir que tu te fasses ta propre config

williamgoulois commented on README.md:
````diff
@@ -1,34 +1,62 @@
 
-The `pages/api` directory is mapped to `/api/*`. Files in this directory are treated as [API routes](https://nextjs.org/docs/api-routes/introduction) instead of React pages.
+```bash
+npm install
````
Il serait temps de passer à yarn ou pnpm je te laisse voir ces options

williamgoulois commented on README.md:
```diff
@@ -1,34 +1,62 @@
-npm run dev
-# or
-yarn dev
+git clone https://github.com/baptistesx/im-lazy-frontend-nextjs
```
Tu as pas encore setup de clé SSH ? 🙂

baptistesx replied:
C'est pour les repos privés les clés SSH non? Celui là est publique

williamgoulois commented on tsconfig.json:
```diff
@@ -13,7 +13,10 @@
     "resolveJsonModule": true,
```
ajoute toi des alias https://medium.com/slackernoon/use-typescript-aliases-to-clean-up-your-import-statements-7210b7ec2af1

baptistesx replied:
ah nice ça je connaissais pas!

williamgoulois commented on tsconfig.json:
```diff
@@ -13,7 +13,10 @@
     "resolveJsonModule": true,
```
Tu devras aussi te faire ta propre config ici pour step up ton TS

williamgoulois commented on utils/functions.tsx:
```diff
@@ -0,0 +1,15 @@
+  }
+
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
```
Si c'est que pour de l'affichage tu dois le faire en css via 
```css
text-transform: capitalize;
```

williamgoulois commented on utils/functions.tsx:
```diff
@@ -0,0 +1,15 @@
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+export const isAdmin = (user: User | null | undefined) =>
+  user?.role === "admin";
```
N'a rien a faire ici doit être dans le provider

williamgoulois replied:
Je préfère que les fonctions is** renvoient un boolean et non boolean | undefined comme ici

baptistesx replied:
isAdmin et isPremium renvoient bien un boolean (test ===), non?

williamgoulois commented on utils/functions.tsx:
```diff
@@ -0,0 +1,15 @@
+export const isAdmin = (user: User | null | undefined) =>
+  user?.role === "admin";
+
+export const isPremium = (user: User | null | undefined) =>
```
Même chose doit être dans le provider

williamgoulois commented on utils/constants.tsx:
```diff
@@ -0,0 +1,14 @@
+
+export const GOOGLE_CLIENT_ID = process.env.GOOGLE_CLIENT_ID;
+
+export const PAYPAL_CLIENT_ID = process.env.PAYPAL_CLIENT_ID;
```
Chacune doit être scopée à son domaine de l'application, si une page fait appel à ce fichier elle a accès à toutes les variables c'est pas terrible en terme de sécu (c'est pour ça d'ailleurs que tu ne peux pas déconstruire `process.env`

williamgoulois commented on utils/constants.tsx:
```diff
@@ -0,0 +1,14 @@
+
+export const PAYPAL_SANDBOX_CLIENT_ID = process.env.PAYPAL_SANDBOX_CLIENT_ID;
+
+export const DEV_MODE = process.env.DEV_MODE;
```
Tu as des variables définies par Next qui suivent si tu es en dev ou en prod je te laisse aller voir la doc 
https://nextjs.org/docs/basic-features/environment-variables

williamgoulois commented on styles/globals.css:
```diff
@@ -4,8 +4,12 @@ body {
 
+body > * {
+  height: 100%;
+}
```
Je ne pense pas que tu aies besoin de ça (ou sinon il faut définir un layout de page mais pas ici)

baptistesx replied:
En fait c'est sur la balise d'id #__next que j'ai besoin de height 100%

![image](https://user-images.githubusercontent.com/9638281/159519616-86903ac3-e717-4371-95af-901d5ae1e46f.png)

du coup je remplace par:
 #__next {
	height: 100%;
}

c'est good?

williamgoulois commented on services/workawayBotApi.tsx:
```diff
@@ -0,0 +1,51 @@
+import api from "./api";
```
Il manque cruellement les types de retours dans toutes ces fonctions

williamgoulois commented on services/workawayBotApi.tsx:
```diff
@@ -0,0 +1,51 @@
+};
+
+export const clearLogs = async (cb: Function) => {
+  await api.axiosApiCall({ url: "clearLogs", method: "get" }).then((res) => {
```
Les urls sont par convention en kebab-case pas camelCase

williamgoulois commented on services/workawayBotApi.tsx:
```diff
@@ -0,0 +1,51 @@
+};
+
+export const getFilesName = async (cb: Function) => {
+  await api.axiosApiCall({ url: "filesName", method: "get" }).then((res) => {
```
pas de mix entre async await et .then()

williamgoulois commented on services/workawayBotApi.tsx:
```diff
@@ -0,0 +1,51 @@
+  });
+};
+
+export const deleteFileByName = async (name: string, cb: Function) => {
```
Function est un type bien trop large c'est mieux que tu types la signature de la fonction si tu la connais

williamgoulois commented on services/workawayBotApi.tsx:
```diff
@@ -0,0 +1,51 @@
+    });
+};
+
+export const startBot = async (data: any, cb: Function) => {
```
Empêche l'utilisation de any si tu veux vraiment progresser en TS

williamgoulois commented on services/workawayBotApi.tsx:
```diff
@@ -0,0 +1,51 @@
+import api from "./api";
+
```
Aucune gestion d'erreur ?

baptistesx replied:
C'est catch dans les composants

williamgoulois commented on services/api.tsx:
```diff
@@ -0,0 +1,24 @@
+  }) {
+    return axios({
+      method,
+      url: `${ENDPOINT}${url}`,
```
Habitue toi à ne pas inclure le / de fin dans l'endpoint

williamgoulois commented on services/api.tsx:
```diff
@@ -0,0 +1,24 @@
+      url: `${ENDPOINT}${url}`,
+      data: body,
+      headers: { "Content-Type": "application/json" },
+      withCredentials: true,
```
séparer 2 clients axios (un avec authent et un sans)

williamgoulois commented on services/api.tsx:
```diff
@@ -0,0 +1,24 @@
+import axios, { Method } from "axios";
+import { ENDPOINT } from "../utils/constants";
+
+const api = {
```
(TS) Ici tu dois utiliser un type générique qui sera le type de tes données de retour (et possiblement le type aussi du body pour éviter le any)

williamgoulois replied:
Dans le cas ou tu as plusieurs back avec des config différentes il faut en créer plusieurs

williamgoulois commented on services/userApi.tsx:
```diff
@@ -0,0 +1,176 @@
+
+export const signOut = async (cb: Function) => {
+  await api.axiosApiCall({ url: "signOut", method: "post" });
+  console.log("well logged out");
```
À enlever ou à wrap autour d'un test si on est bien en env de dev (voir la doc des variables d'env de next)

baptistesx replied:
Pourquoi? Ce call api permet au back de remove le cookie contenant le token de session

williamgoulois commented on services/userApi.tsx:
```diff
@@ -0,0 +1,176 @@
+  const users = await api
+    .axiosApiCall({ url: "users", method: "get" })
+    .then((res) => res.data.users)
+    .catch((err) => {});
```
SI tu catch il faut toujours que tu traites l'erreur ou que tu throw

williamgoulois commented on services/userApi.tsx:
```diff
@@ -0,0 +1,176 @@
+import { Fetcher } from "swr";
```
Encore des mix d'async await et .then .catch il faut que tu choisisses une des deux syntaxes

williamgoulois replied:
Des any et des Function à mieux typer

williamgoulois commented on services/userApi.tsx:
```diff
@@ -0,0 +1,176 @@
+
+export const sendVerificationEmail = async (
+  email: string | undefined,
+  cb: Function
```
tu fais l'appel avec email === undefined ? Il faut que tu le vérifies avant et que cette fonction ne reçoive que `email: string`

williamgoulois commented on schemas/updatePasswordFormSchema.tsx:
```diff
@@ -0,0 +1,18 @@
+  })
+  .required();
+
+export default updatePsswordFormSchema;
```
typo

williamgoulois commented on schemas/updatePasswordFormSchema.tsx:
```diff
@@ -0,0 +1,18 @@
+    currentPassword: yup
+      .string()
+      .min(6, "Password must be at least 6 characters")
+      .max(255)
```
ta bdd supporte bien jusqu'à 255 ?

baptistesx replied:
oui

williamgoulois commented on schemas/signUpFormSchema.tsx:
```diff
@@ -0,0 +1,27 @@
+  .object({
+    name: yup
+      .string()
+      .min(2, "Name must be at least 6 characters")
```
```suggestion
      .min(6, "Name must be at least 6 characters")
```

williamgoulois commented on schemas/infoFormSchema.tsx:
```diff
@@ -0,0 +1,47 @@
+      .min(15, "Minimum age must be at least 15")
+      .max(79, "Maximum age must be under 79")
+      .required("Minimum age range required"),
+    maximumAge: yup
```
Si maximum === minimum ça fait bug ton app ? (juste une question)

baptistesx replied:
Il y a une condition sur maximumAge (moreThan(...) par rapport à minimumAge) qui est trigger pour avoir minimum < maximum

williamgoulois commented on schemas/infoFormSchema.tsx:
```diff
@@ -0,0 +1,47 @@
+      .min(5, "Message subject must be at least 5 characters")
+      .max(50)
+      .required("Message subject is required"),
+    frenchMessage: yup
```
Si dans ta logique du formulaire il y a un french message et un english message ok sinon si c'est french ou english ce n'est pas ici que tu dois gérer les langues

baptistesx replied:
Oui, c'est à l'utilisateur de rentrer un message en francais et un en anglais (qui peuvent différer s'il le souhaite).
Le script du bot enverra le message français ou anglais suivant la nationalité de la personne à mp.

williamgoulois commented on schemas/editUserFormSchema.tsx:
```diff
@@ -0,0 +1,22 @@
+      .required("Email is required"),
+    role: yup
+      .mixed<Role>()
+      .oneOf(Object.values(["admin", "premium", "classic"])),
```
Object.values ne devrait pas être nécessaire ici non ?

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+export interface AlertContextInterface {
+  addAlert: Function;
```
refine

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+  addAlert: Function;
+}
+
+export const SnackbarContext = createContext<AlertContextInterface | null>(
```
initialise à undefined plutôt null est réservé aux valeurs existantes mais vides

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+    </SnackbarContext.Provider>
+  );
+}
+export const useSnackbars = () => useContext(SnackbarContext);
```
Ajoute un guard comme défini ici
https://kentcdodds.com/blog/how-to-use-react-context-effectively#typescript

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+export type Alert = {
+  message: string;
+  severity: AlertColor;
+};
```
Doit être dans son propre composant il n'a rien à faire ici

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+  severity: AlertColor;
+};
+export function SnackBarProvider({ children }: { children: React.ReactNode }) {
+  const [alerts, setAlerts] = useState<Alert[]>([]);
```
Je préfère que tu initialises systématiquement à undefined plutôt que des objets ou tableaux vides

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+  null
+);
+
+const AUTO_DISMISS = 5000;
```
```suggestion
const AUTO_DISMISS_MS = 5000;
```

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+  }, [activeAlertIds]);
+
+  const addAlert = (alert: Alert) => {
+    console.log(alert);
```
🧹

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+import { Alert, AlertColor, Box, Snackbar } from "@mui/material";
+import { createContext, useContext, useEffect, useState } from "react";
+
+export interface AlertContextInterface {
```
C'est un SnackBarProvider ou un AlertProvider ? Il faut que tu respectes le SRP https://medium.com/@roni.shabo/single-responsibility-in-reactjs-9c60e4163862

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+      const timer = setTimeout(
+        () => setAlerts((alerts) => alerts.slice(0, alerts.length - 1)),
+        AUTO_DISMISS
+      );
```
Article génial sur les setInterval (qui est aussi valable pour les setTimeout) je te conseille de l'ajouter à tes custom hooks
https://overreacted.io/making-setinterval-declarative-with-react-hooks/

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+
+  const activeAlertIds = alerts.join(",");
+
+  useEffect(() => {
```
donne un nom à tes fonctions dans le useEffect ça aide vraiment à savoir ce qu'elles font

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+export function SnackBarProvider({ children }: { children: React.ReactNode }) {
+  const [alerts, setAlerts] = useState<Alert[]>([]);
+
+  const activeAlertIds = alerts.join(",");
```
à quoi te sert exactement cette variable ?

baptistesx replied:
rien

williamgoulois commented on providers/SnackbarProvider.tsx:
```diff
@@ -0,0 +1,57 @@
+  };
+
+  return (
+    <SnackbarContext.Provider value={alertContext}>
```
Tu passes une alertContext à un SnackBarContext ça ne te fait pas un petit tilt ? 🙂 
Voir le message sur le single responsibility principle

williamgoulois commented on providers/CustomThemeProvider.tsx:
```diff
@@ -0,0 +1,29 @@
+import { createTheme, ThemeProvider,responsiveFontSizes, Theme as MuiTheme } from "@mui/material/styles";
+import React from "react";
+
+declare module "@emotion/react" {
+  export interface Theme extends MuiTheme {}
+}
```
N'a rien à faire ici doit être dans un fichier .d.ts

williamgoulois commented on providers/CustomThemeProvider.tsx:
```diff
@@ -0,0 +1,29 @@
+}: {
+  children: JSX.Element;
+}) => {
+  let theme = createTheme({
```
Je préfère un const et tu appelles directement responsiveFontSizes(createTheme(...))

williamgoulois replied:
Le let c'est que quand une variable va vraiment changer (donc très rare)

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+
+type AuthStatus = "loading" | "connected" | "not-connected" | "error";
+
+type AuthValue = {
+  user: User | null | undefined;
+  status: AuthStatus;
+  logout: () => void;
+  login: (email: string, password: string, cb: Function) => void;
+  register: (
+    name: string,
+    email: string,
+    password: string,
+    cb: Function
+  ) => void;
+  loginWithGoogle: (token: string, cb: Function) => void;
+  fetchCurrentUser: () => void;
+};
```
Tu dois définir tes différents états possibles avec qqch comme ça :
```tsx
type Connected = {
	user: User
	status: "connected"
}
type NotConnected = {
	user: null
	status: "not-connected"
}
//etc...
```
Cela permet à TS de savoir avec un test `if (status === 'connected')` de savoir que user est de type User

williamgoulois commented on providers/CustomThemeProvider.tsx:
```diff
@@ -0,0 +1,29 @@
+export const CustomThemeProvider = ({
+  children,
+}: {
+  children: JSX.Element;
```
Pourquoi pas ReactNode ?

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+} from "../services/userApi";
+import { useSnackbars } from "./SnackbarProvider";
+
+export type Role = "admin" | "premium" | "classic";
```
J'espère que tu t'en sers comme indication seulement et pas source absolue de vérité, c'est le back qui est le garant du role

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+
+type AuthStatus = "loading" | "connected" | "not-connected" | "error";
+
+type AuthValue = {
```
Sépare le statut de l'auth et les actions dans 2 provider différent (qui peuvent être dans le même fichier) avec deux fonctions
useAuth() et useAuthActions()

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+  user: User | null | undefined;
+  status: AuthStatus;
+  logout: () => void;
+  login: (email: string, password: string, cb: Function) => void;
```
Les Function encore

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+type AuthStatus = "loading" | "connected" | "not-connected" | "error";
+
+type AuthValue = {
+  user: User | null | undefined;
```
Pourquoi pas User | undefined ? null correspond à un "vrai" état ?

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+    cb: Function
+  ) => void;
+  loginWithGoogle: (token: string, cb: Function) => void;
+  fetchCurrentUser: () => void;
```
Ne doit pas être exposée par le provider c'est de la logique interne

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const fetcher = (url: RequestInfo) => getUserSWR();
+
+  const { data: userSWR, mutate, error } = useSWR("user", fetcher);
+  const [user, setUser] = useState<User | null | undefined>(null);
```
Tu te retrouves avec 2 sources de vérité pour ton user => ton state et ton cache de SWR
Tu ne devrais pas te servir de SWR dans ce cas je pense

williamgoulois replied:
À initialiser avec undefined null c'est pour une variable définie mais vide c'est pas la même chose

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+
+  const router = useRouter();
+
+  useEffect(() => {
```
nom fonction useEffect

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const fetcher = (url: RequestInfo) => getUserSWR();
+
+  const { data: userSWR, mutate, error } = useSWR("user", fetcher);
```
Tu ajoutes error alors qu'elles ne sont pas gérées par ton fetcher dans ton api

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+  useEffect(() => {
+    setStatus("loading");
+    fetchCurrentUser();
+    setUser(userSWR);
```
⚠️ tu as déjà un setUser dans ton fetchCurrentUser
Sans vérifier je sais déjà que tu fais plusieurs appels à getUser

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+
+      // TODO: there is probably a better way to get NotSignedInRoutes
+      if (
+        router.route !== "/" &&
+        router.route !== "/auth/sign-in" &&
+        router.route !== "/auth/sign-up" &&
+        router.route !== "/auth/reset-password"
```
Tester `router.pathname.startsWIth(routes.auth)` avec routes un fichier ou tu types toutes tes routes

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+        router.route !== "/auth/sign-up" &&
+        router.route !== "/auth/reset-password"
+      ) {
+        snackbarsService?.addAlert({
+          message: "An error occured while fetching user.",
+          severity: "error",
+        });
+      }
```
Le snackbarService qui fait office d'alert provider 😉

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+    getUser((user: User) => {
+      setUser(user);
+      setStatus("connected");
+    }).catch((err: Error) => {
```
pas besoin de le typer l'error dans tous les catch sera toujours unknown

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+      setStatus("connected");
+
+      snackbarsService?.addAlert({
+        message: !user.lastLogin ? "Welcome to ImLazy app !" : "Welcome back !",
```
Tu devrais avoir un fichier de traductions et utiliser https://www.i18next.com/ pour tout centraliser

williamgoulois replied:
Fais des tests explicites
```suggestion
        message: user.lastLogin === undefined ? "Welcome to ImLazy app !" : "Welcome back !",
```

williamgoulois replied:
Fais des tests explicites
```suggestion
        message: user.lastLogin === undefined ? "Welcome to ImLazy app !" : "Welcome back !",
```

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+      setUser(null);
+      setStatus("not-connected");
+      //TODO: add internet connection checker and customize message error
+      snackbarsService?.addAlert({
```
Il faut que tu wrappes toutes tes actions dans un useMemo et tu test une seule fois que tout est bien initialisé (le router, ton snackbarService etc

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+  );
+};
+
+export const useAuth = () => useContext(AuthContext);
```
ajouter un guard comme expliqué dans le snackbarprovider

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+  };
+
+  return (
+    <AuthContext.Provider
+      value={{
+        user,
+        status,
+        logout,
+        login,
+        loginWithGoogle,
+        register,
+        fetchCurrentUser,
+      }}
+    >
+      {children}
+    </AuthContext.Provider>
```
2 providers
```suggestion
    <AuthContext.Provider
      value={{
        user,
        status,
      }}
    >
    <AuthContextActions.Provider value={{ logout,
        login,
        loginWithGoogle,
        register}}>
      {children}
      </AuthContextActions.Provider>
    </AuthContext.Provider>
```

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+    name: string,
+    email: string,
+    password: string,
+    cb: Function
```
mieux typer tous les cb

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+        severity: "success",
+      });
+
+      router.push("/dashboard");
```
Ce n'est pas la responsabilité du login de déterminer la prochaine page

williamgoulois commented on providers/AuthProvider.tsx:
```diff
@@ -0,0 +1,202 @@
+  lastLogin: Date;
+};
+
+type AuthStatus = "loading" | "connected" | "not-connected" | "error";
```
Ton status error n'est jamais utilisé => en cas d'erreur tu va dans l'état "not-connected"

williamgoulois commented on next.config.js:
```diff
@@ -1,6 +1,13 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
```
Tu as encore pas mal de choses que tu peux custom hésites pas à lire plus la doc

williamgoulois commented on package.json:
```diff
@@ -9,9 +9,25 @@
     "lint": "next lint"
   },
   "dependencies": {
```
certaines dépendances ne devraient être que des devDependancies

williamgoulois reviewed (commented):
Ajoute prettier et editorconfig pour éviter de le refaire à chaque fois
Une CI légère (voir une CD) serait la bienvenue
Ajouter husky pour automatiser le lint et format à chaque commit

baptistesx commented:
Avec Husky, voir :
https://www.codeheroes.fr/2021/10/11/git-lutilisation-des-hooks-avec-husky/
https://www.codeheroes.fr/2020/06/29/git-comment-nommer-ses-branches-et-ses-commits/

baptistesx committed:
ci(config): add git hooks/Husky

Auto commit msg & branch name check

baptistesx committed:
ci(config): add git hooks/Husky

Auto commit msg & branch name check

baptistesx committed:
ci(config): update branch name regex

baptistesx committed:
fix(paypal): update paypal for production

baptistesx committed:
ci(config): setup husky

baptistesx committed:
Merge branch 'chore/integrate-williams-review' of github.com:baptistesx/im-lazy-frontend-nextjs into chore/integrate-williams-review

baptistesx commented:
Ajouter un check auto des console.log

baptistesx committed:
ci(config): setup eslint & editorconfig #10 #11

baptistesx committed:
fix(global): linter errors/warnings

baptistesx committed:
refactor(global): use alises

baptistesx committed:
refactor(global): remove capitalizeFirstLetter()

Replace capitalizeFirstLetter function with sx textTransform prop·

baptistesx committed:
refactor(global): isAdmin & isPremium in provider

baptistesx committed:
refactor(routes): camelCase -> kebab-case

baptistesx committed:
refactor(async): remove all then()

baptistesx committed:
refactor(routes): remove '/' in ENDPOINT

baptistesx committed:
refactor(global): fix typo

baptistesx committed:
refactor(api): test user email

Test user email before calling api with potential undefined var

baptistesx committed:
refactor(routes): move routes in folder

baptistesx committed:
fix(form): reset(data) for ProfileForm

baptistesx committed:
ci(config): aff check & format on pre-commit

baptistesx committed:
refactor(global): global updates on william advice

baptistesx committed:
ci(config): add remove unused imports

baptistesx committed:
refactor(env): refactor env vars

- Remove env vars in next.config.js (old nextjs versions) and use NEXT_PUBLIC in vars name
- Create seperates files for development & production

baptistesx committed:
refactor(env): add eslint rules

- Fix all linter errors

baptistesx committed:
refactor(functions): some functional components

baptistesx committed:
refactor(functions): function type => detailed

baptistesx committed:
ci(rules): add eslint rules

baptistesx committed:
refactor(global): replace any & init to undefined

baptistesx commented:
Pour avoir "Verified" sur les commits: https://gist.github.com/troyfontaine/18c9146295168ee9ca2b30c00bd1b41e

baptistesx committed:
refactor(global): integrate review comments

baptistesx committed:
refactor(auth): isAdmin/isPremium => useAuth

baptistesx committed:
feat(i18n): start implementing i18n

baptistesx committed:
feat(i18n): continue to implement i18n

baptistesx committed:
refactor(json): update i18n files

baptistesx committed:
refactor(clean): run prettier

baptistesx committed:
refactor(i18n): update i18n

baptistesx committed:
refactor(drawer): fix drawers collapse

baptistesx committed:
feat(theme): add theme settings

baptistesx committed:
refactor(i18n): update french translations

baptistesx committed:
refactor(drawer): implement i18n in the drawer

baptistesx committed:
refactor(theme): remove custom themes

baptistesx committed:
refactor(bot): fix bugs

baptistesx committed:
refactor(layout-menu): drawer settings -> navbar

baptistesx committed:
refactor(snackbar): fix snackbar

baptistesx committed:
refactor(global): clean global

baptistesx committed:
refactor(theme): some fixes

baptistesx committed:
refactor(token): logout when token expired

baptistesx committed:
refactor(snackbaProvider): notistack for snackbars

baptistesx committed:
feat(internet-connection): add connection checker

baptistesx committed:
refactor(todos): fix various todos

baptistesx committed:
refactor(workaway-files): update files

Update table (add sorting/selecting)
Request only users file

baptistesx committed:
refactor(bot-layout): some fix

baptistesx committed:
refactor(theme): fix bug

baptistesx committed:
refactor(.d.tsx): remove .t.tsx files

baptistesx committed:
refactor(users): rename files table

baptistesx committed:
refactor(users): upgrade users table

baptistesx committed:
refactor(undefined-case): add undefined check

baptistesx committed:
refactor(bot): improve bot work

baptistesx committed:
refactor(layout): repsonsive Fonts + layout

baptistesx committed:
refactor(global): global cleaning

baptistesx committed:
refactor(build): fix build error + install sharp

baptistesx committed:
refactor(logo): fix logo color

baptistesx committed:
refactor(clean): remove console.log()

baptistesx committed:
refactor(cleaning): some cleaning

baptistesx committed:
refactor(update): minor changes

baptistesx committed:
fix: fix build issues

baptistesx committed:
Merge pull request #16 from baptistesx/chore/integrate-williams-review

Chore/integrate williams review

baptistesx committed:
update default values
